R/sgdnet.R

```diff
@@ -16,6 +16,34 @@
 
 #' Fit a Generalized Linear Model with Elastic Net Regularization
 #'
+#' @section Model families:
+#' Two model families are currently supported: gaussian univariate
+#' regression and binomial logistic regression, the choice of which is made
+#' using the `family` argument.
+#'
+#' The *gaussian* family solves the following objective:
+#'
+#' \deqn{
+#'   \frac{1}{2n} \sum_{i=1}^n (y_i -\beta_0 - x_i^\intercal \beta)^2
+#'   + \lambda \left( \frac{1 - \alpha}{2} ||\beta||_2^2
+#'                    + \alpha||\beta||_1 \right)
+#' }{
+#'   1/(2n) \sum (y -\beta_0 - x^T \beta)^2
+#'   + \lambda [(1 - \alpha)/2 ||\beta||_2^2 + \alpha||\beta||_1]
+#' }
+#'
+#' The *binomial* family solves the following objective:
+#'
+#' \deqn{
+#'   -\frac1n \sum_{i=1}^n \log\Big(1 + e^{-y_i x_i^\intercal \beta + \beta_0}\Big)
+#'   + \lambda \left( \frac{1 - \alpha}{2} ||\beta||_2^2
+#'                    + \alpha||\beta||_1 \right),
+#' }{
+#'   -1/n \sum log[1 + exp(-y x^T \beta + \beta_0)]
+#'   + \lambda [(1 - \alpha)/2 ||\beta||_2^2 + \alpha||\beta||_1],
+#' }
+#' where \eqn{y_i \in {-1, 1}}{y ~ {-1, 1}}.
```

Comment:
So do we want `y` coded as 0/1 or as -1/+1? This should be clear in the docs (below `y` is just described as the response variable)

Reply:
It doesn't matter to me, but it is currently coded as 0/1 in the algorithm, so I'll change the documentation to reflect this.